server/initmodules.py

```diff
@@ -158,6 +158,18 @@ def load_parameter_spec(d, module_version, order):
     pspec.derived_data = d.get('derived-data', False)
     pspec.placeholder = d.get('placeholder', '')
 
+    # For now I will only allow one of them to be active to avoid logically weird stuff.
```

Comment:
Yes, they can definitely contradict. I think I would prefer one "visible_if" field and one "invert_visible_condition" boolean. This seems like it would simplify everything downstream, as well as make it impossible for the visible_if and visible_if_not fields to get into a conflicting state.